Modules/cjkcodecs/cjkcodecs.h

```diff
@@ -149,40 +149,43 @@ static const struct dbcs_map *mapping_list;
         writer->pos += 2;                                                  \
     } while (0)
 
-#define OUTBYTE1(c) \
-    do { ((*outbuf)[0]) = (c); } while (0)
-#define OUTBYTE2(c) \
-    do { ((*outbuf)[1]) = (c); } while (0)
-#define OUTBYTE3(c) \
-    do { ((*outbuf)[2]) = (c); } while (0)
-#define OUTBYTE4(c) \
-    do { ((*outbuf)[3]) = (c); } while (0)
+#define OUTBYTEI(c, i)         \
+    do {                       \
+        unsigned int _c = (c); \
```

Comment:
This can unnecessary widen the argument if it is already more narrow than `int` or truncate higher bits if it is wider than `int`.

Reply:
Yes, I know, but the proper solution is `typeof`, which is not used in Python. Or I could use `assert(0 <= (c) && (c) <= 255);`, but `c` will be evaluated multiple times.

Widening is not harmful here since we check the value immediately afterwards (and all negative values converted to `2**32 - abs(c)` will be out of range). Truncation can lead to missed problems, but it is no worse than the original code which performed implicit truncation to `unsigned char'.